File: src/LeetCode/Algorithms/Find_N_Unique_Integers_Sum_up_to_Zero.java
package LeetCode.Algorithms;

// https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/
public class Find_N_Unique_Integers_Sum_up_to_Zero {
	public static void main(String[] args){
		int n = 5;
		int[] result = sumZero(n);
		
		for (int i : result) {
			System.out.print(i + " ");
		}
		System.out.println();
	}

	public static int[] sumZero(int n) {
		int[] result = new int[n];
		int size = n / 2;
		int index = 0;
		
		while (size > 0) {
			result[index++] = -size;
			result[index++] = size;
			size--;
		}
		
		return result;
	}
}

File: src/LeetCode/Algorithms/JavaScript/Find_N_Unique_Integers_Sum_up_to_Zero.js
/**
 * https://leetcode.com/problems/find-n-unique-integers-sum-up-to-zero/
 * @param {number} n
 * @return {number[]}
 */
var sumZero = function(n) {
	let result = [];
	
	for (let i = 1; i < n; i += 2) {
		result.push(i, -i);
	}
	
	if (n % 2 == 1) {
		result.push(0);
	}
	
	return result;
};

let n = 5;

console.log(sumZero(n));